shotover-proxy/src/frame/cassandra.rs

```diff
@@ -63,6 +82,18 @@ impl CassandraFrame {
         }
     }
 
+    // Count the amonut of queries in this `CassandraFrame`, this will either be the count of all queries in a BATCH statement or 1 for all other queries
+    pub(crate) fn get_query_count(&self) -> Result<NonZeroU32> {
+        Ok(match &self.operation {
+            CassandraOperation::Batch(bytes) => {
+                let short_bytes = &bytes[1..3];
+                let short = u16::from_be_bytes(short_bytes.try_into()?);
+                NonZeroU32::new(short.into()).unwrap()
+            }
+            _ => nonzero!(1u32),
```

Comment:
this seems wrong.
operations other than Batch and Query should have count 0.
And is it gauranteed that query will always contain exactly one query.
I dont see that specified in the spec anywhere.

Reply:
After discussion with conor:
* The name is misleading the desired functionality is to count messages and batch should be considered as each query being multiple sub messages.
* By sending messages like `INSERT INTO foo.foo (field) VALUES (1); INSERT INTO foo.foo (field) VALUES (2);` and `SELECT * FROM system.peers; SELECT * FROM system.peers;` I can see that these get split and sent in seperate messages.
    + It might be worth checking what happens if we manually construct a message with multiple queries in a single QUERY message but I cant imagine it working as there would be no way for the protocol to properly respond to such a message.

Reply:
Ended up hacking shotover to send my cursed query:
![image](https://user-images.githubusercontent.com/5120858/157815250-131d58ab-b6b5-465e-9fd4-336213ff9703.png)
And I can confirm that cassandra rejects it.